packages/query-core/src/infiniteQueryObserver.ts

```diff
@@ -23,7 +23,7 @@ type InfiniteQueryObserverListener<TData, TError> = (
 
 export class InfiniteQueryObserver<
   TQueryFnData = unknown,
-  TError = unknown,
+  TError extends Error = Error,
```

Comment:
why `extends Error` ? This will make it impossible for users to opt out of this, for example if they `throw "some error occurred"`, their `TError` is `string`, which doesn't `extend Error` ❗

Reply:
yes, I got that but sometimes we need to narrow the type, I changed it, can you take a look? if it's not better can you suggest a solution?